packages/eas-build-job/src/metadata.ts

```diff
@@ -2,6 +2,17 @@ import Joi from 'joi';
 
 import { Workflow } from './common';
 
+export enum FingerprintSourceType {
+  'GCS' = 'GCS',
+  'PATH' = 'PATH',
+  'URL' = 'URL',
+}
+
+export type FingerprintSource =
+  | { type: FingerprintSourceType.GCS; bucketKey: string }
+  | { type: FingerprintSourceType.PATH; path: string }
+  | { type: FingerprintSourceType.URL; url: string };
```

Comment:
I believe you need to modify `index.ts` explicitly to export these

https://github.com/expo/eas-build/blob/9cd384ab3f62624587e7c6e469bc894a9cd5b561/packages/eas-build-job/src/index.ts#L16